### PR TITLE
[FIX] discuss: prevent call public page test race condition

### PR DESCRIPTION
**Pull request by ThanhDodeurOdoo**
Before this commit, it was possible that a bus message containing a `mail.message.rtc.session` record update reached the client before the response to the `/mail/rtc/channel/join_call` route.

Since the condition for the test step is that there is a call (triggered by the presence of a rtc session) while the condition for the error thrown is that there is no `selfSession` (which is obtained as the response of the http request), one condition could be satisfied without the other.

### CODE REVIEW

**Review by alexkuhn — APPROVED** [review]
@robodoo delegate+

File: addons/mail/models/mail_ice_server.py [hunk 1]
```diff
@@ -42,6 +44,7 @@ def _get_ice_servers(self):
             (account_sid, auth_token) = get_twilio_credentials(self.env)
             if account_sid and auth_token:
                 url = f'https://api.twilio.com/2010-04-01/Accounts/{account_sid}/Tokens.json'
+                _logger.info("Contacting twilio at: %s", url)
```
Comment (seb-odoo) [inline addons/mail/models/mail_ice_server.py:47]: Why is there an extra log? That looks irrelevant to fixing the race condition.

File: addons/mail/static/tests/tours/discuss_channel_call_public_tour.js [hunk 2]
```diff
@@ -22,7 +22,8 @@ registry.category("web_tour.tours").add("discuss_channel_call_public_tour.js", {
         {
             content: "Check that the call has started",
             trigger: ".o-discuss-Call",
-            run() {
+            async run() {
+                await new Promise((r) => setTimeout(r, 250));
```
Comment (seb-odoo) [inline addons/mail/static/tests/tours/discuss_channel_call_public_tour.js:26]: What makes 250 enough?

Comment (alexkuhn) [reply addons/mail/static/tests/tours/discuss_channel_call_public_tour.js:26]: IMO I'd prefer a DOM state mutation (UI visible or not... former preferred ofc) or a selfSessionProm awaiting.

The `await 250` is not the best, but we can relatively deduce from the run below that `o-discuss-Call` and `selfSession` requires some time to match. The DOM state mutation awaiting feels artificial, but would be more robust to race conditions.

File: addons/mail/models/mail_ice_server.py (hunk 1 above)
Comment (ThanhDodeurOdoo) [reply addons/mail/models/mail_ice_server.py:47]: xdo wanted to check it during multi run test on runbot, I will remove it before merge

File: addons/mail/static/tests/tours/discuss_channel_call_public_tour.js (hunk 2 above)
Comment (ThanhDodeurOdoo) [reply addons/mail/static/tests/tours/discuss_channel_call_public_tour.js]: Instead of checking the state of the rtc_service, I replaced with a check on the `Disconnect` button from the call context menu, it should only be there if `rtc_service.state.channel` is the current channel, which is equivalent to checking the `selfSession` as channel is set at the same time (when the RPC returns)

File: addons/mail/static/tests/tours/discuss_channel_call_public_tour.js [hunk 3]
```diff
@@ -22,11 +22,10 @@ registry.category("web_tour.tours").add("discuss_channel_call_public_tour.js", {
...
             content: "Check that the call has started",
             trigger: ".o-discuss-Call",
-            run() {
-                if (!odoo.__WOWL_DEBUG__.root.env.services["discuss.rtc"]?.selfSession) {
-                    console.error("The call should have started.");
-                }
-            },
+        },
+        {
+            content: "Check that the Disconnect button is visible",
```
Comment (alexkuhn) [inline addons/mail/static/tests/tours/discuss_channel_call_public_tour.js:27]: `Check that current user is in call ("disconnect" button visible)`

This step would feel less random by re-wording that way

**Review by alexkuhn — COMMENTED** [review]
LGTM (already delegated+)

**Comment by ThanhDodeurOdoo** [comment]
@robodoo r+
